core/src/main/java/hudson/model/RunParameterDefinition.java

```diff
@@ -92,7 +92,7 @@ private RunParameterDefinition(String name, String projectName, String runId, St
     public ParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof RunParameterValue) {
             RunParameterValue value = (RunParameterValue) defaultValue;
-            return new RunParameterDefinition(getName(), value.getRunId(), getDescription(), getFilter());
+            return new RunParameterDefinition(getName(), getProjectName(), value.getRunId(), getDescription(), getFilter());
```

Comment:
Shouldn't this use `value.getProjectName()`?

Reply:
RunParameterValue does not have a getProjectName()... I think it is fair to derive it from the Definition, since that's the place where we fetch the build filter.